var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -196,6 +200,51 @@ def url_for_version(self, version):
         url = "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-{0}/hdf5-{1}/src/hdf5-{1}.tar.gz"
         return url.format(version.up_to(2), version)
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('-showconfig', output=str, error=str)
+        match = re.search(r'HDF5 Version: ([\d.]+)', output)
+        return match.group(1) if match else None
+
+    @classmethod
+    def determine_variants(cls, exes, version):
+        for exe in exes:
+            variants = ''
+            output = Executable(exe)('-showconfig', output=str, error='str')
+
+            def determine_boolean_feature(feature, feature_variant):
```

Comment:
If CMake is being used in the parent package, these options are directly usable after the find_package call as cmake variables.